File: spring-activemq/src/main/java/com/springactivemq/controller/ApiController.java
package com.springactivemq.controller;

import com.springactivemq.config.JMSProducer;
import com.springactivemq.config.MessageInfo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(value = "/api")
public class ApiController {
    @Autowired
    private JMSProducer jmsProducer;

    @PostMapping(value = "/sendMessage")
    public void sendMessage() {
        jmsProducer.sendMessage("jms message!");
    }

    @PostMapping(value = "/handleObj")
    public void handleObj() {
        MessageInfo message = MessageInfo.builder()
                .batchCode(2)
                .classCode("1121245")
                .clientCode("1001001")
                .clientMemCode(23)
                .build();
        jmsProducer.handleObj(message);
    }
}

File: spring-redis/src/main/kotlin/com/springredis/controller/ApiController.kt
package com.springredis.controller

import com.fasterxml.jackson.databind.ObjectMapper
import com.springredis.service.SalaryService
import com.springredis.service.UserService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.cache.CacheManager
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping(value = ["/api"])
class ApiController {

    @Autowired
    lateinit var userService: UserService

    @Autowired
    lateinit var salaryService: SalaryService

    private val logger = LoggerFactory.getLogger(this.javaClass)

    @Autowired
    @Qualifier("customObjectMapper")
    lateinit var mapper: ObjectMapper

    @Autowired
    lateinit var cacheManager: CacheManager

    @GetMapping("/testCache")
    fun testCache() {
        logger.info("------------------ demo @Cacheable --------------------");
        logger.info("find user with id = 1: {}", userService.findUserById(1));
        logger.info("find user with id = 1: {}", userService.findUserById(1));
        logger.info("find user with id = 2: {}", userService.findUserById(2));
        logger.info("find user with id = 2: {}", userService.findUserById(2));
        logger.info("------------------ demo @CacheEvict --------------------");
        userService.clearCache();
        logger.info("find user with id = 1: {}", userService.findUserById(1));
        logger.info("find user with id = 2: {}", userService.findUserById(2));
        logger.info("------------------ demo @CachePut --------------------");
        logger.info("reload and find user with id = 1: {}", userService.reloadAndFindUserById(1));
        logger.info("find user with id = 1: {}", userService.findUserById(1));
        logger.info("find user with id = 2: {}", userService.findUserById(2));
    }

    @GetMapping("/getEmployeeWithSalary")
    fun getEmployeeWithSalary(@RequestParam salary: Int, @RequestParam gender: String): List<*> {
        return salaryService.getEmployeeWithSalary(salary, gender)
    }

    @PostMapping("/clearSingleCache")
    fun clearSingleCache(@RequestParam cacheName: String, @RequestParam cacheKey: String) {
        cacheManager.getCache(cacheName)!!.evict(cacheKey)
    }

    @PostMapping("/clearAllCache")
    fun clearAllCache(@RequestParam cacheName: String) {
        cacheManager.getCache(cacheName)!!.clear()
    }

}